### PR TITLE
`safeDepositAddress` for `fetchDepositAddresses` 

### PR DESCRIPTION
**Pull request by ttodua**
`safeDepositAddress` to be used in `fetchDepositAddresses` (through `parseDepositAddress`).
open for further improvements.

### CODE REVIEW

**Comment by kroitor** [comment]
Let's add the support for `currency` structure and propagate it through `safeDepositAddress`.

File: js/base/Exchange.js [hunk 1]
```diff
@@ -1394,6 +1394,16 @@ module.exports = class Exchange {
         return indexed ? this.indexBy (result, 'currency') : result;
     }
 
+    safeDepositAddress (depositAddress) {
+        return this.extend({
```
Comment (kroitor) [inline js/base/Exchange.js:1398]: If possible, please use consistent spacing here `this.extend({` > `this.extend ({` to keep it uniform across the file.

File: js/base/Exchange.js [hunk 2]
```diff
@@ -1394,6 +1394,16 @@ module.exports = class Exchange {
...
+            'currency': undefined,
+            'address': undefined,
+            'tag': undefined,
+            'network': undefined,
+            'info': {},
```
Comment (kroitor) [inline js/base/Exchange.js:1403]: The `info` must be `undefined`.

File: js/base/Exchange.js [hunk 3]
```diff
@@ -1394,6 +1394,16 @@ module.exports = class Exchange {
         return indexed ? this.indexBy (result, 'currency') : result;
     }
 
+    safeDepositAddress (depositAddress) {
+        return this.extend({
+            'currency': undefined,
```
Comment (kroitor) [inline js/base/Exchange.js:1399]: Let's use a `currency['code']` here.

**Comment by ttodua** [comment]
yeap, I have to return back to the `safeX` PR's from today, to address them.

**Comment by ttodua** [comment]
ready to be merged

**Pull request merged**